File: src/main/java/dev/vrba/slack/tenor/bot/service/slack/dto/SlackViewUpdateRequest.java
package dev.vrba.slack.tenor.bot.service.slack.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.NonNull;
import io.micronaut.serde.annotation.Serdeable;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@Serdeable
@AllArgsConstructor
public class SlackViewUpdateRequest {
    @NonNull
    @JsonProperty("view_id")
    private final String id;

    @NonNull
    @JsonProperty("view")
    private final SlackView view;
}

File: src/main/kotlin/dev/vrba/slack/service/slack/dto/SlackViewUpdateRequest.kt
package dev.vrba.slack.service.slack.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class SlackViewUpdateRequest(
    @JsonProperty("view_id")
    val id: String,
    @JsonProperty("view")
    val view: SlackView,
)
